### PR TITLE
Update attention.py, using SDPA by default

### PR DESCRIPTION
**Pull request by popcornell**
This PR enables the use of torch native `scaled_dot_product_attention` by DEFAULT when it is available. 
`scaled_dot_product_attention` in new torch versions uses by default `flash_attention` so we do not need to install it separately.
This allow by default massive speedups with new torch versions. 
See https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html

⚠️ since this is default based on torch version my guess is that it can break inference with pretrained models. 
So I am not really sure about this.

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks!
There is an issue in the CI (comes from the object type) https://github.com/espnet/espnet/actions/runs/15634791507/job/44047283802?pr=6149

Can you check it?

**Comment by popcornell** [comment]
I also discovered that this change breaks espnet1 transformer encoder: `espnet/nets/pytorch_backend/transformer/encoder.py` since the mask would be different. 
Should I go forward ?

**Comment by sw005320** [comment]
I see. Thanks for checking the issues.

1. we will remove espnet1 in the espnet3 branch. So, we may make this PR to espnet3.
2. @wanchichen told me that some codes (e.g., espnet2 SE) use espent1 transformer (@Emrys365, is it correct?). So, we may solve this part anyway...

**Comment by Emrys365** [comment]
> 2. @wanchichen told me that some codes (e.g., espnet2 SE) use espent1 transformer (@Emrys365, is it correct?). So, we may solve this part anyway...

Yes, I checked the SE related code and found the following espnet1-related dependencies:

```python
# used in 
#   espnet2/enh/extractor/td_speakerbeam_extractor.py
#   espnet2/enh/layers/dnn_wpe.py
#   espnet2/enh/layers/mask_estimator.py
from espnet.nets.pytorch_backend.nets_utils import make_pad_mask

# used in
#   espnet2/enh/separator/conformer_separator.py
#   espnet2/enh/separator/transformer_separator.py
from espnet.nets.pytorch_backend.nets_utils import make_non_pad_mask

# used in
#   espnet2/enh/layers/dptnet.py
from espnet.nets.pytorch_backend.nets_utils import get_activation

# used in
#  espnet2/enh/layers/mask_estimator.py
#  espnet2/enh/separator/dan_separator.py
#  espnet2/enh/separator/dpcl_e2e_separator.py
#  espnet2/enh/separator/dpcl_separator.py
#  espnet2/enh/separator/rnn_separator.py
from espnet.nets.pytorch_backend.rnn.encoders import RNN, RNNP

# used in
#   espnet2/enh/separator/conformer_separator.py
from espnet.nets.pytorch_backend.conformer.encoder import Encoder as ConformerEncoder

# used in
#   espnet2/enh/separator/transformer_separator.py
from espnet.nets.pytorch_backend.transformer.embedding import (
    PositionalEncoding,
    ScaledPositionalEncoding,
)
from espnet.nets.pytorch_backend.transformer.encoder import (
    Encoder as TransformerEncoder,
)
```

**Comment by popcornell** [comment]
Then probably safer to just move the PR to point to ESPnet3 development branch.

**Comment by sw005320** [comment]
I think so.

@Masao-Someki, can you add it to the roadmap? (SPDA and deprecate espent1 transformer)

**Comment by Masao-Someki** [comment]
@popcornell You can change base to espnet3/. Note that if you want to run CI, you need to change some CI files

e.g., 
https://github.com/espnet/espnet/pull/6139/files#diff-b13dff138f65e577647f126f573d36d77fe805c8018d819d439da401f913456c

**Comment by Fhrozen** [comment]
Due to the removal of espnet, this PR is no longer able to merge into the master branch. Please submit a new PR with the fixed applied to the new master branch.
